packages/holocron/src/holocronModule.jsx

```diff
@@ -55,11 +55,16 @@ export default function holocronModule({
         }
       }
 
-      componentWillReceiveProps(nextProps) {
-        if (shouldModuleReload && shouldModuleReload(this.props, nextProps)) {
-          const newLoadCount = this.state.loadCount + 1;
+      // ignoring to support deprecated componentWillReceiveProps.
+      // This needs to be removed to support React17
+      // eslint-disable-next-line camelcase
+      UNSAFE_componentWillReceiveProps(nextProps) {
+        const { loadCount } = this.state;
+        const { props } = this;
+        if (shouldModuleReload && shouldModuleReload(props, nextProps)) {
+          const newLoadCount = loadCount + 1;
           this.setState({ status: 'loading', loadCount: newLoadCount });
-          this.initiateLoad(newLoadCount, nextProps);
+          this.initiateLoad(newLoadCount, nextProps, loadCount);
```

Comment:
this looks like an unrelated change. why was it done?

Reply:
line `80` was causing a ```react/destructuring-assignment``` eslint error to be raised and destructuring it to `loadCount` caused me to compare the same vars, decided to pass this a param to the `initiateLoad` function but this might have caused the behaviour of the function to change.